templates/nextflow-aurora-mysql.yaml

```
@@ -103,6 +106,18 @@ Resources:
         PasswordLength: 16
       KmsKeyId: !Sub ${AuroraKeyStack.Outputs.Key}
 
+  AuroraMasterSecretRotationSchedule:
+    Condition: UseSnapshotFalse
+    Type: AWS::SecretsManager::RotationSchedule
+    Properties:
+      SecretId: !Ref AuroraMasterSecret
+      HostedRotationLambda:
+        RotationType: MySQLSingleUser
+        MasterSecretArn: !Ref AuroraMasterSecret
+        MasterSecretKmsKeyArn: !Sub ${AuroraKeyStack.Outputs.KeyArn}
```

When I was testing the latest commit, I ran into an error. I was able to overcome it by switching from `MasterSecretKmsKeyArn` to `KmsKeyArn`. It seems that the `MasterSecret*` parameters are only relevant with the alternating user setup. 

```suggestion
        KmsKeyArn: !Sub ${AuroraKeyStack.Outputs.KeyArn}
```

---

Yup, i have confirmed that deployment only worked when removing Master secret with just the `KmsKeyArn`.  After deploying successfully I manually triggered the  secret rotation but I got a lambda timeout error (timeout was set to 30 secs) ..

```
START RequestId: e598024b-e5d3-4619-94b8-e919fcf1a621 Version: $LATEST
[ERROR]	2023-04-06T15:50:07.301Z	e598024b-e5d3-4619-94b8-e919fcf1a621	setSecret: Unable to log into database with previous, current, or pending secret of secret arn arn:aws:secretsmanager:us-east-1:035458030717:secret:nextflow-aurora-mysql-MasterSecret-08Z0bn
[ERROR] ValueError: Unable to log into database with previous, current, or pending secret of secret arn arn:aws:secretsmanager:us-east-1:035458030717:secret:nextflow-aurora-mysql-MasterSecret-08Z0bn
Traceback (most recent call last):
  File "/var/task/lambda_function.py", line 75, in lambda_handler
    set_secret(service_client, arn, token)
  File "/var/task/lambda_function.py", line 195, in set_secret
    raise ValueError("Unable to log into database with previous, current, or pending secret of secret arn %s" % arn)
END RequestId: e598024b-e5d3-4619-94b8-e919fcf1a621
REPORT RequestId: e598024b-e5d3-4619-94b8-e919fcf1a621	Duration: 40683.06 ms	Billed Duration: 40684 ms	Memory Size: 128 MB	Max Memory Used: 72 MB
```

I then manually increased the timeout to 1 minute, i'm not sure how you would set the timeout in cloudformation.  Then I manually re-triggered the secret rotation again.  This time the lambda failed to access the database.  The error..
```
START RequestId: e598024b-e5d3-4619-94b8-e919fcf1a621 Version: $LATEST
[ERROR]	2023-04-06T15:50:07.301Z	e598024b-e5d3-4619-94b8-e919fcf1a621	setSecret: Unable to log into database with previous, current, or pending secret of secret arn arn:aws:secretsmanager:us-east-1:035458030717:secret:nextflow-aurora-mysql-MasterSecret-08Z0bn
[ERROR] ValueError: Unable to log into database with previous, current, or pending secret of secret arn arn:aws:secretsmanager:us-east-1:035458030717:secret:nextflow-aurora-mysql-MasterSecret-08Z0bn
Traceback (most recent call last):
  File "/var/task/lambda_function.py", line 75, in lambda_handler
    set_secret(service_client, arn, token)
  File "/var/task/lambda_function.py", line 195, in set_secret
    raise ValueError("Unable to log into database with previous, current, or pending secret of secret arn %s" % arn)
END RequestId: e598024b-e5d3-4619-94b8-e919fcf1a621
REPORT RequestId: e598024b-e5d3-4619-94b8-e919fcf1a621	Duration: 40683.06 ms	Billed Duration: 40684 ms	Memory Size: 128 MB	Max Memory Used: 72 MB	
```